### ISSUE TITLE
WARNING: method definition for #computeWavelets and cwt

### ISSUE THREAD

**Issue by shayandavoodii** [JuliaDynamics/CausalityTools.jl#176]
Hi.
Today I installed this package successfully. Now I tried to import it, But it threw this bunch of warnings:
```julia
julia> using CausalityTools
[ Info: Precompiling CausalityTools [5520caf5-2dd7-5c5d-bfcb-a00e56ac49f7]
WARNING: method definition for #computeWavelets#6 at C:\Users\Shayan\.julia\packages\Wavelets\ANOxi\src\mod\WT.jl:660 declares type variable S but does not use it.
WARNING: method definition for #computeWavelets#7 at C:\Users\Shayan\.julia\packages\Wavelets\ANOxi\src\mod\WT.jl:664 declares type variable S but does not use it.
WARNING: method definition for #computeWavelets#8 at C:\Users\Shayan\.julia\packages\Wavelets\ANOxi\src\mod\WT.jl:670 declares type variable S but does not use it.
WARNING: method definition for cwt at C:\Users\Shayan\.julia\packages\Wavelets\ANOxi\src\mod\Transforms.jl:216 declares type variable U but does not use it.
```

**Comment by kahaaga** [comment]
Hey @shayandavoodii! Thanks for letting us know.

This warning comes from the Wavelets.jl package (which CausalityTools depends on). The first line of the warning tells you that there is something happening in line 660 of the `src\mod\WT.jl` file in the Wavelets package. Finding the Wavelets.jl repo on github and navigating to that file, you'll see the following (I hope the link works):

https://github.com/JuliaDSP/Wavelets.jl/blob/master/src/mod/WT.jl#:~:text=function%20computeWavelets(Y%3A%3ATuple,end

The function definition on that line is 

```julia 
computeWavelets(Y::Tuple, c::CFW{W}; T=Float64) where {S<:Real, W<:WT.WaveletBoundary}
```

The code authors have type-annotated the arguments `Y` and `c`, demanding that `Y` is a `Tuple` and `c` is a `CWT{<:WaveletBoundary}`. However, they also specify a parameter `S <: Real` which is not used to type-annotate any of the input arguments. That's why you're seeing this warning. As far as I can see, this has zero impact on the code.

TLDR: You can safely disregard these warnings. It's just Julia's compiler reminding the code authors (and users) that there is something going on in the code that shouldn't happen. In this case, the issue happens to be harmless. 

If it's annoying to have these warning, consider making opening an issue or submitting a PR to [Wavelets.jl](https://github.com/JuliaDSP/Wavelets.jl) to fix it :) If you decide to do so, you should include a new minimally reproducible example (installing only Wavelets.jl in a clean Julia environment, not including this package).

**Comment by shayandavoodii** [comment]
@kahaaga, Thanks for this complete explanation. I got your point. Surely I'll open an issue in the related package. Thanks for the energy and time you've put into this explanation. This means the world to me.
